pytest_rts/utils/common.py

```diff
@@ -65,42 +68,41 @@ def tests_from_changed_srcfiles(diff_dict, files, db_helper):
 
         if any(
             [
-                not db_helper.mapping_line_exists(file_id, line_id)
-                for line_id in new_lines
+                not mappinghelper.line_exists(file_id, line_number)
+                for line_number in new_lines
             ]
         ):
             files_to_warn.append(filename)
 
-        tests = db_helper.query_tests_srcfile(changed_lines, file_id)
+        test_set.update(testgetter.get_tests_from_srcfiles(changed_lines, file_id))
 
-        for test in tests:
-            test_set.add(test)
     return test_set, changed_lines_map, new_line_map, files_to_warn
 
 
-def split_changes(changed_files, db_helper):
+def split_changes(changed_files, mappinghelper):
     """Split given changed files into changed testfiles and source code files"""
-    changed_tests = []
-    changed_sources = []
-    db_test_files, db_src_files = db_helper.get_testfiles_and_srcfiles()
+    changed_testfiles = []
+    changed_srcfiles = []
+
+    db_testfiles = mappinghelper.testfiles
+    db_srcfiles = mappinghelper.srcfiles
 
     for changed_file in changed_files:
-        for srcfile in db_src_files:
+        for srcfile in db_srcfiles:
             path_to_file = srcfile[1]
             if changed_file == path_to_file:
-                changed_sources.append(srcfile)
-        for testfile in db_test_files:
+                changed_srcfiles.append(srcfile)
```

Comment:
This code needs refactoring; it's too nested and does complicated data manipulation. Helper function might help; or some other refactoring.